rust/arrow/src/array/array_list.rs

```diff
@@ -295,6 +298,185 @@ impl fmt::Debug for FixedSizeListArray {
+                        Time64MicrosecondType,
+                        $list_builder
+                    )
+                }
+                DataType::Time64(TimeUnit::Nanosecond) => {
+                    build_empty_list_array_with_primitive_items!(
+                        Time64NanosecondType,
+                        $list_builder
+                    )
+                }
+                DataType::Duration(TimeUnit::Second) => {
+                    build_empty_list_array_with_primitive_items!(
+                        DurationSecondType,
+                        $list_builder
+                    )
+                }
+                DataType::Duration(TimeUnit::Millisecond) => {
+                    build_empty_list_array_with_primitive_items!(
+                        DurationMillisecondType,
+                        $list_builder
+                    )
+                }
+                DataType::Duration(TimeUnit::Microsecond) => {
+                    build_empty_list_array_with_primitive_items!(
+                        DurationMicrosecondType,
+                        $list_builder
+                    )
+                }
+                DataType::Duration(TimeUnit::Nanosecond) => {
+                    build_empty_list_array_with_primitive_items!(
+                        DurationNanosecondType,
+                        $list_builder
+                    )
+                }
+                DataType::Timestamp(TimeUnit::Second, _) => {
+                    build_empty_list_array_with_primitive_items!(
+                        TimestampSecondType,
+                        $list_builder
+                    )
+                }
+                DataType::Timestamp(TimeUnit::Millisecond, _) => {
+                    build_empty_list_array_with_primitive_items!(
+                        TimestampMillisecondType,
+                        $list_builder
+                    )
+                }
+                DataType::Timestamp(TimeUnit::Microsecond, _) => {
+                    build_empty_list_array_with_primitive_items!(
+                        TimestampMicrosecondType,
+                        $list_builder
+                    )
+                }
+                DataType::Timestamp(TimeUnit::Nanosecond, _) => {
+                    build_empty_list_array_with_primitive_items!(
+                        TimestampNanosecondType,
+                        $list_builder
+                    )
+                }
+                DataType::Utf8 => {
+                    build_empty_list_array_with_non_primitive_items!(
+                        StringBuilder,
+                        $list_builder
+                    )
+                }
+                DataType::Binary => {
+                    build_empty_list_array_with_non_primitive_items!(
+                        BinaryBuilder,
+                        $list_builder
+                    )
+                }
+                _ => Err(ArrowError::Unsupported(format!(
```

Comment:
What `ArrowError` should I throw on implemented? todo!? Or let's declare this one?

Thanks